pybind/pybind11
# Add GHA and several test fixes

This adds a basic GHA test suite. Features:

* macOS, Windows, and Linux, all versions of Python, default C++ standard.

Test related fixes:

* Fix for PyPy2 tests not passing
* CMake passthrough fix for cmake test, fixes #2295 (will possibly improve later during the CMake cleanup)
* Fix warning for Python 2 + C++14+ (will possibly improve later during the CMake cleanup and help user code - test only for now)
* IDE-based build systems (MSVC) were not able to run the embed test - fixed.
* Adding a virtual environment directory with `env` in the name is now ignored.
* Fixed Python 3.9+ warning (upstream, see https://github.com/python/cpython/pull/21592 )

Future plans:

* Fix the Windows Python 3.8+ builds
* Add multiple C++ version checks
* Add docker-based checks too, for CUDA & LLVM versions
* Add a `-DDOWNLOAD_EIGEN` for CMake, add Eigen testing here.
* Add docs check and other specialty builds too
* If the matrix gets too big, maybe split up.
* Once this has fairly good overlap with old CI, start dropping old CI.

This should be "rebased & merged".
